core/src/test/java/org/apache/struts2/util/FastByteArrayBufferTest.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.struts2.util;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.CodingErrorAction;
+
+import junit.framework.TestCase;
+
+
+/**
+ * Basic tests of FastByteArrayOutputStream
+ */
+public class FastByteArrayBufferTest extends TestCase {
+
+    final String utf16_String = new String("Standard string with accented characters \u00E7\u00E8\u00E9 in the middle");
+    String utf8_String = null;
+    String iso8859_1_String = null;
+
+
+    protected void setUp() throws Exception {
+        super.setUp();
+        
+        try {
+            if (utf8_String == null) {
+                utf8_String = new String(utf16_String.getBytes("UTF-8"), "UTF-8");
```

Comment:
You could use `public String(byte bytes[], Charset charset)` with `java.nio.charset.StandardCharsets`.

According to the Javadoc "These charsets are guaranteed to be available on every implementation of the Java platform."

Benefit: No need to handle `UnsupportedEncodingException`.

Reply:
Fair point, as using the static `StandardCharsets` would allow for immediate assignment in the initializer block.  That would permit elimination of the `setup()` method.